Lib/test/eintrdata/eintr_tester.py

```diff
@@ -39,13 +39,16 @@ class EINTRBaseTest(unittest.TestCase):
 
     # delay for initial signal delivery
     signal_delay = 0.1
+    # signal handler run time duration
+    signal_duration = 0.05
     # signal delivery periodicity
     signal_period = 0.1
     # default sleep time for tests - should obviously have:
     # sleep_time > signal_period
     sleep_time = 0.2
 
     def sighandler(self, signum, frame):
+        time.sleep(self.signal_duration)
```

Comment:
I am not convinced about this test. It relies on timing and this is very error prone on different platforms. We have a lot of buildbot failing all the time because the tests rely on timing. Here, the signal delivery can be affected by lots of things (they are not processed instantly, as the Python signal handlers only set up a flag for later execution).

Check out https://github.com/python/cpython/pull/10413 for a more extended rationale.

Reply:
I agree, I wasn't 100% happy with it either. I will take a look and revise the test. Thanks.

Reply:
Please remove this sleep. It seems like a very bad idea to sleep in a thread handler. Moreover, I don't see the purpose of sleeping!?